clients/src/main/java/org/apache/kafka/clients/ClientUtils.java

```diff
@@ -172,7 +173,37 @@ public static NetworkClient createNetworkClient(AbstractConfig config,
                 null,
                 new DefaultHostResolver(),
                 throttleTimeSensor,
-                clientTelemetrySender);
+                clientTelemetrySender,
+                Function.identity());
```

Comment:
i think we are going pretty deep to access the metadata error, i wonder if we could simplify the pattern.

we can probably directly access the metadata error in the background thread and send it to the client, in this case we don't need to create a separated wrapper around the updater.

wdyt if we just create a metadata error manager in the consumer network thread level and poll the metadata error directly?

Reply:
Are you suggesting creating a RequestManager for Metadata Errors?

In my understanding, the role of a RequestManager is to manage outgoing requests. 
However, looking at the MembershipManagerImpl, it seems like a good idea. 
By adopting this approach, we could propagate Metadata Error while maintaining the structure of the new Consumer.

Reply:
@philipnee I've applied it